src/framework/mlt_chain.c

```diff
@@ -395,6 +395,7 @@ void mlt_chain_close(mlt_chain self)
             mlt_link_close(base->links[i]);
         free(base->links);
         mlt_profile_close(base->source_profile);
+        mlt_producer_close(base->source);
```

Comment:
And what about source_parameters?

Reply:
Good catch. I fixed that with a force push.